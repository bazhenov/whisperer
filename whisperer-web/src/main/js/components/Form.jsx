import React, {Component} from 'react';

export default class Form extends Component {

	constructor(props) {
		super(props);
		this.handleInputChange = this.handleInputChange.bind(this);
	}

	handleInputChange(e) {
		this.props.updateCurrentConnectionParams(e.target.name, e.target.value);
	}

	render() {
		const { isListening, connectionParams, startListening, stopListening } = this.props;
		return <div className="well">
			<form onSubmit={(e) => e.preventDefault()} >
				<fieldset disabled={isListening}>
					<div className="form-group">
						<label htmlFor="mdcKeyInput">MDC key name</label>
						<input type="text" name="key" onChange={this.handleInputChange} className="form-control"
									 id="mdcKeyInput" value={connectionParams.key} />
					</div>
					<div className="form-group">
						<label htmlFor="mdcValueInput">Expected value</label>
						<input type="text" name="value" onChange={this.handleInputChange} className="form-control"
									 id="mdcValueInput" value={connectionParams.value} />
					</div>
					<div className="form-group" id="log-prefix-input-div">
						<label htmlFor="logPrefixInput">Log prefix</label>
						<input type="text" name="prefix" onChange={this.handleInputChange} className="form-control"
									 id="logPrefixInput" value={connectionParams.prefix} />
					</div>
					<div className="form-group" id="severity-input-div">
						<label htmlFor="severityInput">Severity</label>
						<select className="form-control" id="severityInput" value={connectionParams.severity} name="severity"
										onChange={this.handleInputChange}>
							<option value="TRACE">TRACE</option>
							<option value="DEBUG">DEBUG</option>
							<option value="INFO">INFO</option>
							<option value="WARN">WARN</option>
							<option value="ERROR">ERROR</option>
						</select>
					</div>
				</fieldset>
			</form>
			{isListening ?
				<a href="#" onClick={(e) => { stopListening(); e.preventDefault() }} className="connection-form-button btn btn-danger">STOP</a> :
				<a href="#" onClick={(e) => { startListening(); e.preventDefault() }} className="connection-form-button btn btn-success">LISTEN</a>}
		</div>
	}
}