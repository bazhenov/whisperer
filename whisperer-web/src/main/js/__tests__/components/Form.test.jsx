import React from "react";
import Form from "../../components/Form.jsx";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import { connectionParams } from '../fixtures'

describe('Form', () => {
	const props = {
		connectionParams: connectionParams,
		isListening: false,
		updateCurrentConnectionParams: () => {}
	};

	it('rendering', () => {
		let tree = renderer.create(
			<Form { ...props } />
		).toJSON();
		expect(tree).toMatchSnapshot();

		tree = renderer.create(
			<Form { ...props } isListening={true} />
		).toJSON();
		expect(tree).toMatchSnapshot()
	});

	describe('form changes handler', () => {
		it('key changed', done => {
			const checkKeyChanged = (field, inputId) => {
				const updateCurrentConnectionParams = (key, value) => {
					expect(key).toBe(field);
					expect(value).toBe('qwe');
					done();
				};
				const wrapper = mount(
					<Form { ...props } updateCurrentConnectionParams={updateCurrentConnectionParams} />
				);
				wrapper.find(inputId).node.value = 'qwe';
				wrapper.find(inputId).simulate('change');
			};
			checkKeyChanged('key', '#mdcKeyInput');
			checkKeyChanged('value', '#mdcValueInput');
			checkKeyChanged('prefix', '#logPrefixInput');
		});

		it('severity changed', done => {
			const updateCurrentConnectionParams = (key, value) => {
				expect(key).toBe('severity');
				expect(value).toBe('WARN');
				done();
			};
			const wrapper = mount(
				<Form { ...props } updateCurrentConnectionParams={updateCurrentConnectionParams} />
			);
			wrapper.find('.severity-button').first().simulate('click');
		});
	});
});