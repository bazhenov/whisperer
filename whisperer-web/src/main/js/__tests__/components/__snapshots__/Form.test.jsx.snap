// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form rendering 1`] = `
<div>
  <h3>
    Connection parameters
  </h3>
  <form
    onSubmit={[Function]}
  >
    <fieldset
      disabled={false}
    >
      <div
        className="form-group"
      >
        N
        <label
          htmlFor="mdcKeyInput"
        >
          MDC key name
        </label>
        <input
          className="form-control"
          id="mdcKeyInput"
          name="key"
          onChange={[Function]}
          type="text"
          value="clientId"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="mdcValueInput"
        >
          Expected value
        </label>
        <input
          className="form-control"
          id="mdcValueInput"
          name="value"
          onChange={[Function]}
          type="text"
          value="some_ring"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="logPrefixInput"
        >
          Log prefix
        </label>
        <input
          className="form-control"
          id="logPrefixInput"
          name="prefix"
          onChange={[Function]}
          type="text"
          value="com.farpost.search"
        />
      </div>
    </fieldset>
    <div
      className="form-group"
    >
      <div>
        <label>
          Severity
        </label>
      </div>
      <span
        className="choosed-severity label label-primary"
      >
        INFO
      </span>
      <a
        className="severity-button"
        href="#"
        onClick={[Function]}
      >
        WARN
      </a>
      <a
        className="severity-button"
        href="#"
        onClick={[Function]}
      >
        ERROR
      </a>
      <a
        className="severity-button"
        href="#"
        onClick={[Function]}
      >
        DEBUG
      </a>
      <a
        className="severity-button"
        href="#"
        onClick={[Function]}
      >
        TRACE
      </a>
    </div>
  </form>
</div>
`;

exports[`Form rendering 2`] = `
<div>
  <h3>
    Connection parameters
  </h3>
  <form
    onSubmit={[Function]}
  >
    <fieldset
      disabled={true}
    >
      <div
        className="form-group"
      >
        N
        <label
          htmlFor="mdcKeyInput"
        >
          MDC key name
        </label>
        <input
          className="form-control"
          id="mdcKeyInput"
          name="key"
          onChange={[Function]}
          type="text"
          value="clientId"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="mdcValueInput"
        >
          Expected value
        </label>
        <input
          className="form-control"
          id="mdcValueInput"
          name="value"
          onChange={[Function]}
          type="text"
          value="some_ring"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="logPrefixInput"
        >
          Log prefix
        </label>
        <input
          className="form-control"
          id="logPrefixInput"
          name="prefix"
          onChange={[Function]}
          type="text"
          value="com.farpost.search"
        />
      </div>
    </fieldset>
    <div
      className="form-group"
    >
      <div>
        <label>
          Severity
        </label>
      </div>
      <span
        className="choosed-severity label label-primary"
      >
        INFO
      </span>
      <span
        className="severity-button"
      >
        WARN
      </span>
      <span
        className="severity-button"
      >
        ERROR
      </span>
      <span
        className="severity-button"
      >
        DEBUG
      </span>
      <span
        className="severity-button"
      >
        TRACE
      </span>
    </div>
  </form>
</div>
`;
